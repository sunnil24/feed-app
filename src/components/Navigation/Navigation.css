.navbar-nav {
  list-style: none;
  padding: 0 0 0 16px;
  font-size: 12px;
}

.navbar-item {
  display: inline-block;
}

.navbar-link {
  text-decoration: none;
  color: #000;
}

.navbar-item:after {
  content: "|";
  color: #000;
  padding: 0 4px;
}

.navbar-item:last-child:after {
  content: "";
}

.active.navbar-link {
  color: #fff;
}
