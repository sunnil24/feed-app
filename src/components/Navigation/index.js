import React from "react";
import { NavLink } from "react-router-dom";

// styles
import "./Navigation.css";

// data
import navItems from "../../data/navigation";

const Navigation = () => {
  return (
    <nav className="navbar">
      <ul className="navbar-nav">
        {navItems.map((item) => {
          const { displayName, url, title } = item;
          return (
            <li className="navbar-item">
              <NavLink className="navbar-link" to={url} title={title}>
                {displayName}
              </NavLink>
            </li>
          );
        })}
      </ul>
    </nav>
  );
};

export default Navigation;
