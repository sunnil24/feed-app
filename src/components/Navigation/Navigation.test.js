import React from "react";
import { shallow } from "enzyme";

import Navigation from "./index";

let wrapped = shallow(<Navigation />);

describe("<Navigation />", () => {
  it("should render the Title Component correctly", () => {
    expect(wrapped).toMatchSnapshot();
  });
});
