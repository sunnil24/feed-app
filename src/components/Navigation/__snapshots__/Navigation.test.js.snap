// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render the Title Component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="navbar-nav"
      >
        <li
          className="navbar-item"
        >
          <ForwardRef
            className="navbar-link"
            title="Top"
            to="/"
          >
            Top
          </ForwardRef>
        </li>
        <li
          className="navbar-item"
        >
          <ForwardRef
            className="navbar-link"
            title="New"
            to="/some-url"
          >
            New
          </ForwardRef>
        </li>
      </ul>,
      "className": "navbar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="navbar-item"
          >
            <ForwardRef
              className="navbar-link"
              title="Top"
              to="/"
            >
              Top
            </ForwardRef>
          </li>,
          <li
            className="navbar-item"
          >
            <ForwardRef
              className="navbar-link"
              title="New"
              to="/some-url"
            >
              New
            </ForwardRef>
          </li>,
        ],
        "className": "navbar-nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ForwardRef
              className="navbar-link"
              title="Top"
              to="/"
            >
              Top
            </ForwardRef>,
            "className": "navbar-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Top",
              "className": "navbar-link",
              "title": "Top",
              "to": "/",
            },
            "ref": null,
            "rendered": "Top",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "NavLink",
              "propTypes": Object {
                "activeClassName": [Function],
                "activeStyle": [Function],
                "aria-current": [Function],
                "className": [Function],
                "exact": [Function],
                "innerRef": [Function],
                "isActive": [Function],
                "location": [Function],
                "onClick": [Function],
                "replace": [Function],
                "strict": [Function],
                "style": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            },
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ForwardRef
              className="navbar-link"
              title="New"
              to="/some-url"
            >
              New
            </ForwardRef>,
            "className": "navbar-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "New",
              "className": "navbar-link",
              "title": "New",
              "to": "/some-url",
            },
            "ref": null,
            "rendered": "New",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "NavLink",
              "propTypes": Object {
                "activeClassName": [Function],
                "activeStyle": [Function],
                "aria-current": [Function],
                "className": [Function],
                "exact": [Function],
                "innerRef": [Function],
                "isActive": [Function],
                "location": [Function],
                "onClick": [Function],
                "replace": [Function],
                "strict": [Function],
                "style": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            },
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="navbar-nav"
        >
          <li
            className="navbar-item"
          >
            <ForwardRef
              className="navbar-link"
              title="Top"
              to="/"
            >
              Top
            </ForwardRef>
          </li>
          <li
            className="navbar-item"
          >
            <ForwardRef
              className="navbar-link"
              title="New"
              to="/some-url"
            >
              New
            </ForwardRef>
          </li>
        </ul>,
        "className": "navbar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="navbar-item"
            >
              <ForwardRef
                className="navbar-link"
                title="Top"
                to="/"
              >
                Top
              </ForwardRef>
            </li>,
            <li
              className="navbar-item"
            >
              <ForwardRef
                className="navbar-link"
                title="New"
                to="/some-url"
              >
                New
              </ForwardRef>
            </li>,
          ],
          "className": "navbar-nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ForwardRef
                className="navbar-link"
                title="Top"
                to="/"
              >
                Top
              </ForwardRef>,
              "className": "navbar-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Top",
                "className": "navbar-link",
                "title": "Top",
                "to": "/",
              },
              "ref": null,
              "rendered": "Top",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "NavLink",
                "propTypes": Object {
                  "activeClassName": [Function],
                  "activeStyle": [Function],
                  "aria-current": [Function],
                  "className": [Function],
                  "exact": [Function],
                  "innerRef": [Function],
                  "isActive": [Function],
                  "location": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "strict": [Function],
                  "style": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              },
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ForwardRef
                className="navbar-link"
                title="New"
                to="/some-url"
              >
                New
              </ForwardRef>,
              "className": "navbar-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "New",
                "className": "navbar-link",
                "title": "New",
                "to": "/some-url",
              },
              "ref": null,
              "rendered": "New",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "NavLink",
                "propTypes": Object {
                  "activeClassName": [Function],
                  "activeStyle": [Function],
                  "aria-current": [Function],
                  "className": [Function],
                  "exact": [Function],
                  "innerRef": [Function],
                  "isActive": [Function],
                  "location": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "strict": [Function],
                  "style": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              },
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
